Mathlib/Analysis/Calculus/ArcLengthReparametrization.lean

```diff
@@ -0,0 +1,291 @@
+/-
+Copyright (c) 2025 Michael Novak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Novak, Aaron Liu.
+-/
+import Mathlib.Analysis.Calculus.ContDiff.Basic
+import Mathlib.Analysis.Calculus.ContDiff.RCLike
+import Mathlib.Analysis.Calculus.Deriv.MeanValue
+import Mathlib.Analysis.Calculus.InverseFunctionTheorem.Deriv
+import Mathlib.Analysis.Calculus.IteratedDeriv.Defs
+import Mathlib.MeasureTheory.Integral.IntervalIntegral.FundThmCalculus
+import Mathlib.Order.BourbakiWitt
+
```

Comment:
I'm also surprised to see the `BourbakiWitt` import; my intuition is that it surely can be replaced. But we can revisit this later, once all other kinds in your code are ironed out.